File: Ruby/bee 2462.rb
times = gets.split(" ")

pa_H, pa_M = times[0].split(":").map(&:to_i)
cb_H, cb_M = times[1].split(":").map(&:to_i)
pb_H, pb_M = times[2].split(":").map(&:to_i)
ca_H, ca_M = times[3].split(":").map(&:to_i)

pa = pa_H*60 + pa_M
cb = cb_H*60 + cb_M
pb = pb_H*60 + pb_M
ca = ca_H*60 + ca_M

voo_A = (cb - pa + 1440) % 1440
voo_B = (ca - pb + 1440) % 1440

duracao = ((voo_A + voo_B) % 1440) / 2
fuso_minutos = voo_A-duracao

if fuso_minutos <= -750
    fuso_minutos += 1440
elsif fuso_minutos > 750
    fuso_minutos -= 1440
end

puts duracao.to_s + " " + (fuso_minutos/60).to_s

File: C/bee 2462.c
#include <stdio.h>
int main() {
    int paH, cbH, pbH, caH, paM, cbM, pbM, caM;

    scanf("%d:%d %d:%d %d:%d %d:%d", &paH, &paM, &cbH, &cbM, &pbH, &pbM, &caH, &caM);

    int pa = paH*60 + paM;
    int cb = cbH*60 + cbM;
    int pb = pbH*60 + pbM;
    int ca = caH*60 + caM;

    int vooA = (cb - pa + 1440) % 1440;
    int vooB = (ca - pb + 1440) % 1440;

    int duracao = ((vooA + vooB) % 1440)/2;

    int fusoMinutos = vooA - duracao;

    if(fusoMinutos <= -720) {
        fusoMinutos += 1440;
    } else if(fusoMinutos > 720) {
        fusoMinutos -= 1440;
    }

    printf("%d %d\n", duracao, fusoMinutos/60);

    
    return 0;
}
